cmd/kraft/pkg/list/list.go

```diff
@@ -147,7 +149,18 @@ func (opts *List) Run(cmd *cobra.Command, args []string) error {
 	defer iostreams.G(ctx).StopPager()
 
 	cs := iostreams.G(ctx).ColorScheme()
-	table := tableprinter.NewTablePrinter(ctx)
+	var table tableprinter.TablePrinter
+
+	switch strings.ToLower(opts.Output) {
+	case "json":
+		table = tableprinter.NewTablePrinter(ctx, tableprinter.OutputFormatJSON)
+	case "yaml":
+		table = tableprinter.NewTablePrinter(ctx, tableprinter.OutputFormatYAML)
+	case "table":
+		table = tableprinter.NewTablePrinter(ctx, tableprinter.OutputFormatTable)
+	default:
+		return fmt.Errorf("Output format %s is Invalid", opts.Output)
+	}
```

Comment:
We're re-instantiating the structure here. Instead, instantiate a slice of options, e.g.:
```go
var topts []tableprinter.TablePrinterOption
switch strings.ToLower(opts.Output) {
case "json":
  topts = append(topts, tableprinter. WithOutputFormat(tableprinter.OutputFormatJSON))
// ...
}
printer, err := tableprinter.NewTablePrinter(ctx, topts...)
```

Reply:
Done 👍🏽